app/controllers/RegularContributions.scala

```diff
@@ -70,32 +71,74 @@ class RegularContributions(
       )
     }
 
-  def status(jobId: String): Action[AnyContent] = authenticatedAction().async { implicit request =>
+  def displayFormMaybeAuthenticated(): Action[AnyContent] =
+    maybeAuthenticatedAction(recurringIdentityClientId).async { implicit request =>
+      request.user.fold {
+        val uatMode = testUsers.isTestUserOptionalAuth(request)
+        Future(
+          monthlyContributionsPage(None, uatMode)
+        )
+      } { fUser =>
```

Comment:
There's a lot of folding going on here which makes it a little hard to follow - can we split this into some smaller functions?

Reply:
👍